src/main/java/com/google/sps/servlets/MatchServlet.java

```diff
@@ -13,23 +13,32 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
 
 /** Servlet responsible for creating new saved matches AND listing matches. */
 @WebServlet("/matches")
 public class MatchServlet extends HttpServlet {
 
-  private static MatchRepository testRepository;
-  private static User testUser;
-
   @Override
-  public void init() {
+  public void init(ServletConfig config) throws ServletException {
+    super.init(config);
+
     NonPersistentMatchRepository repository = new NonPersistentMatchRepository();
-    testUser = repository.addTestData();
-    testRepository = repository;
+    User testUser = repository.addTestData();
+    MatchRepository testRepository = repository;
+
+    //Set MatchRepository and Current User as ServletContext so that they can be accessed by all servlets
+    getServletContext().setAttribute("matchRepository", testRepository);
+    getServletContext().setAttribute("currentUser", testUser);
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    ServletContext servletContext = getServletContext();
+    MatchRepository testRepository = (MatchRepository) servletContext.getAttribute("matchRepository");
+    User testUser = (User) servletContext.getAttribute("currentUser");
```

Comment:
Here is where you can look up the logged in user via the UserService API. Feel free to defer this for a future pull request.

Reply:
Sounds good. I will be updating this in one of my next pull requests :)